### PR TITLE
remove ssz encoding for CommitteeMember and Share

### LINKED ISSUE #419: We encode Operator and Share even though we don't pass them on the wire
Could it be because of we use the encoding to register validators?
If so this should be documented

### PR DESCRIPTION
**Pull request by alan-ssvlabs**
Fixes #419 

ssz encoding for CommitteeMember and Share are defined but not used, therefore this PR removes the encoding functions.

For CommitteeMember encoding no tests are designed, therefore only the functions are removed. 
For Share encoding, there are tests defined for this function. The tests are also removed as the function itself is actually unused.

### CODE REVIEW

File: types/spectest/run_test.go [hunk 1]
```diff
@@ -79,12 +79,12 @@ func TestJson(t *testing.T) {
...
-				require.NoError(t, err)
-				typedTest := &share.EncodingTest{}
-				require.NoError(t, json.Unmarshal(byts, &typedTest))
-				typedTest.Run(t)
+			// case reflect.TypeOf(&share.EncodingTest{}).String():
+			// 	byts, err := json.Marshal(test)
+			// 	require.NoError(t, err)
+			// 	typedTest := &share.EncodingTest{}
+			// 	require.NoError(t, json.Unmarshal(byts, &typedTest))
+			// 	typedTest.Run(t)
```
Comment (MatheusFranco99) [inline types/spectest/run_test.go:87]: Since we are removing this test type, you can actually delete these lines :)

**Review by MatheusFranco99 — CHANGES_REQUESTED** [review]
Good job! I left just a little request

File: types/spectest/run_test.go (hunk 1 above)
Comment (alan-ssvlabs) [reply types/spectest/run_test.go:87]: You are right, removed

**Review by alan-ssvlabs — COMMENTED** [review]
remove commented code

**Review by MatheusFranco99 — APPROVED** [review]
Good job

**Review by GalRogozinski — APPROVED** [review]

**Review by y0sher — APPROVED** [review]
We are encoding shares for the database using SSZ in the implementation, but I guess we'll have to implement it in the implementation since its not used here in the spec.

**Pull request merged**